### PR TITLE
Get Routes with type frontend decouple

### PR DESCRIPTION
**Pull request by dinhlongviolin1**
Related to Avaiga/taipy-designer#45

Sample return value `[page_name, page_type][]`
image

### CODE REVIEW

File: frontend/taipy-gui/base/src/app.ts [hunk 1]
```diff
@@ -89,9 +92,11 @@ export class TaipyApp {
         this.clientId = "";
         this.context = "";
         this.appId = "";
+        this.routes = undefined;
         const id = getLocalStorageValue(TAIPY_CLIENT_ID, "");
         sendWsMessage(this.socket, "ID", TAIPY_CLIENT_ID, id, id, undefined, false);
         sendWsMessage(this.socket, "AID", "connect", "", id, undefined, false);
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/base/src/app.ts:98]: Can we not send one INIT message instead of 3 ?

**Review by FredLL-Avaiga — APPROVED** [review]

File: frontend/taipy-gui/base/src/app.ts [hunk 2]
```diff
@@ -6,29 +6,31 @@ import { Socket, io } from "socket.io-client";
...
+export type OnReloadHandler = (taipyApp: TaipyApp, removedChanges: ModuleData) => void;
+type Route = [string, string];
 
 export class TaipyApp {
     socket: Socket;
     _onInit: OnInitHandler | undefined;
     _onChange: OnChangeHandler | undefined;
     _onNotify: OnNotifyHandler | undefined;
-    _onReload: onReloadHandler | undefined;
+    _onReload: OnReloadHandler | undefined;
```
Comment (FabienLelaquais) [inline frontend/taipy-gui/base/src/app.ts:20]: 🙂

File: frontend/taipy-gui/base/src/utils.ts [hunk 3]
```diff
@@ -17,16 +17,16 @@ interface AlertMessage extends WsMessage {
 
 const initWsMessageTypes = ["ID", "AID", "GMC"];
 
-export const initSocket = (socket: Socket, appManager: TaipyApp) => {
+export const initSocket = (socket: Socket, taipyApp: TaipyApp) => {
```
Comment (FabienLelaquais) [inline frontend/taipy-gui/base/src/utils.ts:20]: 🙂

**Review by FabienLelaquais — APPROVED** [review]
Perfect.
Loved the renaming.

**Pull request merged**